src/plugins/intel_cpu/tests/functional/single_layer_tests/classes/transpose.cpp

```diff
@@ -0,0 +1,121 @@
+
+using namespace InferenceEngine;
+using namespace CPUTestUtils;
+using namespace ngraph::helpers;
+using namespace ov::test;
+
+namespace CPULayerTestsDefinitions {
+std::string TransposeLayerCPUTest::getTestCaseName(testing::TestParamInfo<TransposeLayerCPUTestParamSet> obj) {
+    Precision netPrecision;
+    InputShape inputShapes;
+    std::vector<size_t> inputOrder;
+    std::string targetDevice;
+    CPUSpecificParams cpuParams;
+    std::map<std::string, std::string> additionalConfig;
+    std::tie(inputShapes, inputOrder, netPrecision, targetDevice, additionalConfig, cpuParams) = obj.param;
+
+    std::ostringstream result;
+    result << "IS=" << CommonTestUtils::partialShape2str({inputShapes.first}) << "_";
+    result << "TS=(";
+    for (const auto& shape : inputShapes.second) {
+        result << CommonTestUtils::vec2str(shape) << "_";
+    }
+    result << ")_";
+    result << "inputOrder=" << CommonTestUtils::vec2str(inputOrder) << "_";
+    result << "netPRC=" << netPrecision.name() << "_";
+    result << "trgDev=" << targetDevice;
+    result << CPUTestsBase::getTestCaseName(cpuParams);
+    return result.str();
+}
+
+void TransposeLayerCPUTest::SetUp() {
+    Precision netPrecision;
+    InputShape inputShapes;
+    std::vector<size_t> inputOrder;
+    CPUSpecificParams cpuParams;
+    std::map<std::string, std::string> additionalConfig;
+    std::tie(inputShapes, inputOrder, netPrecision, targetDevice, additionalConfig, cpuParams) = this->GetParam();
+    configuration.insert(additionalConfig.begin(), additionalConfig.end());
+
+    inType = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+    outType = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+
+    std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+
+    selectedType = makeSelectedTypeStr("unknown", inType);
+
+    init_input_shapes({inputShapes});
+
+    auto params = ngraph::builder::makeDynamicParams(inType, {inputDynamicShapes[0]});
+
+    const auto inputOrderOp =
+        std::make_shared<ov::op::v0::Constant>(ov::element::i64, ov::Shape({inputOrder.size()}), inputOrder);
+    const auto transpose = std::make_shared<ov::op::v1::Transpose>(params[0], inputOrderOp);
+    transpose->get_rt_info() = getCPUInfo();
+    const ov::ResultVector results{std::make_shared<ov::op::v0::Result>(transpose)};
+
+    function = std::make_shared<ngraph::Function>(results, params, "TransposeLayerCPUTest");
+    functionRefs = ngraph::clone_function(*function);
+}
+
+TEST_P(TransposeLayerCPUTest, CompareWithRefs) {
+    run();
+    CheckPluginRelatedResults(compiledModel, "Transpose");
+}
+
+namespace Transpose {
+const std::vector<InferenceEngine::Precision>& netPrecisionsPerChannels() {
+    static const std::vector<InferenceEngine::Precision> netPrecisionsPerChannels = {Precision::I8, Precision::FP32};
+    return netPrecisionsPerChannels;
+}
+
+const std::vector<InputShape>& staticInputShapes4DC16() {
```

Comment:
isn't it actually dynamic? :)
here and below

Reply:
fixed